modules/Rubrics/moduleFunctions.php

```diff
@@ -240,128 +242,244 @@ function rubricView($guid, $connection2, $gibbonRubricID, $mark, $gibbonPersonID
+                    $cellCount = 1 ;
+                    foreach ($cells[$row['gibbonRubricRowID']] AS $cell) {
+                        $visualise = false ;
+                        foreach ($columns as $column) {
+                            if ($column['gibbonRubricColumnID'] == $cell['gibbonRubricColumnID']) {
+                                $visualise = true ;
+                            }
                         }
 
-                        $col->addContent($historicalContent)
-                            ->addClass('historical')
-                            ->prepend('<b><u>' . __('Total Occurences:') . ' ' . $countHistorical . '</u></b><br/>')
-                            ->append(($countHistorical > 7)? '<b>'.__('Older occurrences not shown...').'</b>' : '')
-                            ->append('<span class="cellID" data-cell="' . $cell['gibbonRubricCellID'] . '"></span>');
+                        if ($visualise) {
+                            foreach ($contexts as $entry) {
+                                if ($entry['gibbonRubricCellID'] == $cell['gibbonRubricCellID']) {
+                                    $means[$row['gibbonRubricRowID']]['cumulative'] += $cellCount;
+                                    $means[$row['gibbonRubricRowID']]['denonimator'] ++; 
+                                }
+                            }
+                            $cellCount++;
+                        }
                     }
                 }
-            }
 
-            if ($mark == true) {
-                $output .= "<script type='text/javascript'>";
-                $output .= '$(document).ready(function(){';
-                $output .= '$(".markableCell").parent().click(function(){';
-                    $output .= "var mode = '';";
-                    $output .= "var cellID = $(this).find('.cellID').data('cell');";
-                    $output .= "if ($(this).hasClass('rubricCellHighlight') == false ) {";
-                        $output .= "$(this).addClass('rubricCellHighlight');";
-                        $output .= "mode = 'Add';";
-                    $output .= '} else {';
-                        $output .= "$(this).removeClass('rubricCellHighlight');";
-                        $output .= "mode = 'Remove';";
-                    $output .= '}';
-                    $output .= 'var request=$.ajax({ url: "'.$_SESSION[$guid]['absoluteURL'].'/modules/Rubrics/rubrics_data_saveAjax.php", type: "GET", data: {mode: mode, gibbonRubricID : "' . $gibbonRubricID.'", gibbonPersonID : "'.$gibbonPersonID.'", gibbonRubricCellID : cellID, contextDBTable : "'.$contextDBTable.'",contextDBTableID : "'.$contextDBTableID.'"}, dataType: "html"});';
-                    $output .= '});';
-                $output .= '});';
-                $output .= '</script>';
-            }
+                $output .= "<div id='canvasHolder' style='width:100%;'>";
+                    $output .= "<canvas id='birdsCanvas'></canvas>";
```

Comment:
Birds? 🤔

Reply:
I used some sample code, and then never cleared it up. Just changed it to "canvas" now.